direct/nn/unet/unet_3d.py

```diff
@@ -0,0 +1,439 @@
+
+    def __init__(self, in_channels: int, out_channels: int, dropout_probability: float):
+        """Inits :class:`ConvBlock3D`.
+
+        Parameters
+        ----------
+        in_channels : int
+            Number of channels in the input tensor.
+        out_channels : int
+            Number of channels produced by the convolutional layers.
+        dropout_probability : float
+            Dropout probability applied after convolutional layers.
+        """
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.dropout_probability = dropout_probability
+
+        self.layers = nn.Sequential(
+            nn.Conv3d(in_channels, out_channels, kernel_size=3, padding=1, bias=False),
+            nn.InstanceNorm3d(out_channels),
+            nn.LeakyReLU(negative_slope=0.2, inplace=True),
+            nn.Dropout3d(dropout_probability),
+            nn.Conv3d(out_channels, out_channels, kernel_size=3, padding=1, bias=False),
+            nn.InstanceNorm3d(out_channels),
+            nn.LeakyReLU(negative_slope=0.2, inplace=True),
+            nn.Dropout3d(dropout_probability),
+        )
+
+    def forward(self, input_data: torch.Tensor) -> torch.Tensor:
+        """Performs the forward pass of :class:`ConvBlock3D`..
+
+        Parameters
+        ----------
+        input_data : torch.Tensor
+            Input data.
+
+        Returns
+        -------
+        torch.Tensor
+        """
+        return self.layers(input_data)
+
+
+class TransposeConvBlock3D(nn.Module):
+    """3D U-Net Transpose Convolutional Block."""
+
+    def __init__(self, in_channels: int, out_channels: int):
+        """Inits :class:`TransposeConvBlock3D`.
+
+        Parameters
+        ----------
+        in_channels : int
+            Number of channels in the input tensor.
+        out_channels : int
+            Number of channels produced by the convolutional layers.
+        """
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+
+        self.layers = nn.Sequential(
+            nn.ConvTranspose3d(in_channels, out_channels, kernel_size=2, stride=2, bias=False),
+            nn.InstanceNorm3d(out_channels),
+            nn.LeakyReLU(negative_slope=0.2, inplace=True),
+        )
+
+    def forward(self, input_data: torch.Tensor) -> torch.Tensor:
+        """Performs the forward pass of :class:`TransposeConvBlock3D`.
+
+        Parameters
+        ----------
+        input_data : torch.Tensor
+            Input data.
+
+        Returns
+        -------
+        torch.Tensor
+        """
+        return self.layers(input_data)
+
+
+class UnetModel3d(nn.Module):
+    """PyTorch implementation of a 3D U-Net model.
+
+    This class defines a 3D U-Net architecture consisting of down-sampling and up-sampling layers with 3D convolutional
+    blocks. This is an extension of Unet2dModel, but for volumes.
+    """
+
```

Comment:
Maybe a reference where it has been used? If it's an extension, you can link to the class in the documentation

Reply:
Referenced to our documentation